internal/providers/system/system.go

```diff
@@ -30,12 +32,26 @@ import (
 )
 
 const (
-	baseFilename = "base.ign"
 	userFilename = "user.ign"
 )
 
 func FetchBaseConfig(logger *log.Logger) (types.Config, report.Report, error) {
-	return fetchConfig(logger, baseFilename)
+	baseDirectory := "base.d"
+	// Assuming we get only single directory which is either
+	// a `base.d` or a `base.${PLATFORM_ID}.d`.
```

Comment:
Hmm, I wonder if the implementation might be better suited as either:

- merging both `base.d` & `base.${PLATFORM_ID}.d`

or

- copying the platform stubs into the normal `base.d`

Because we can control merge ordering via file names the individual platform stubs can always come later than the absolute base *COS bits & override as needed so we don't need to duplicate all of the normal stubs that we want into each individual platform config dir.